pkg/controller/config/config_controller.go

```diff
@@ -228,6 +253,101 @@ func (r *ReconcileConfig) reconcileInstall(req reconcile.Request, cfg *op.Config
 		return reconcile.Result{}, err
 	}
 
+	err = r.updateStatus(cfg, op.ConfigCondition{Code: op.AppliedPipeline, Version: flag.TektonVersion})
+	return reconcile.Result{}, err
+}
+
+func (r *ReconcileConfig) validatePipeline(req reconcile.Request, cfg *op.Config) (reconcile.Result, error) {
+	log := requestLogger(req, "validate-pipeline")
+	err := r.updateStatus(cfg, op.ConfigCondition{Code: op.WaitingPipelineValidation, Version: flag.TektonVersion})
+	if err != nil {
+		log.Error(err, "failed to set status")
+		return reconcile.Result{}, err
+	}
+
+	log.Info("validating pipelines")
+	pplnCtrl, err := validate.Deployment(context.TODO(),
+		r.client,
+		flag.PipelineControllerName,
+		cfg.Spec.TargetNamespace,
+	)
+	if err != nil {
+		log.Error(err, "failed to validate pipeline deployment")
+		_ = r.updateStatus(cfg, op.ConfigCondition{
+			Code:    op.ErrorStatus,
+			Details: err.Error(),
+			Version: flag.TektonVersion})
+		return reconcile.Result{}, err
+	}
+	pplnWebhook, err := validate.Deployment(context.TODO(),
+		r.client,
+		flag.PipelineWebhookName,
+		cfg.Spec.TargetNamespace,
+	)
+	if err != nil {
+		log.Error(err, "failed to validate pipeline deployment")
+		_ = r.updateStatus(cfg, op.ConfigCondition{
+			Code:    op.ErrorStatus,
+			Details: err.Error(),
+			Version: flag.TektonVersion})
+		return reconcile.Result{}, err
+	}
+	if !pplnCtrl || !pplnWebhook {
+		log.Info("pipeline controller or/and pipeline webhook is not up")
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: 5 * time.Second,
+		}, nil
+	}
+
+	// NOTE: manifest when updating (not installing) already installed resources
+	// modifies the `cfg` but does not refersh it, hence refresh manually
+	if err := r.refreshCR(cfg); err != nil {
+		log.Error(err, "status update failed to refresh object")
+		return reconcile.Result{}, err
+	}
+
+	err = r.updateStatus(cfg, op.ConfigCondition{Code: op.ValidatedPipeline, Version: flag.TektonVersion})
+	return reconcile.Result{}, err
+}
+
+func (r *ReconcileConfig) reconcileAddons(req reconcile.Request, cfg *op.Config) (reconcile.Result, error) {
+	log := requestLogger(req, "install-pipeline-addons")
+
+	err := r.updateStatus(cfg, op.ConfigCondition{Code: op.InstallingAddons, Version: flag.TektonVersion})
+	if err != nil {
+		log.Error(err, "failed to set status")
+		return reconcile.Result{}, err
+	}
+
+	if err := transformManifest(cfg, &r.addons); err != nil {
+		log.Error(err, "failed to apply manifest transformations on pipeline-addons")
+		// ignoring failure to update
+		_ = r.updateStatus(cfg, op.ConfigCondition{
+			Code:    op.ErrorStatus,
+			Details: err.Error(),
+			Version: flag.TektonVersion})
+		return reconcile.Result{}, err
+	}
+
+	if err := r.addons.ApplyAll(); err != nil {
+		log.Error(err, "failed to apply addons yaml manifest")
+		// ignoring failure to update
+		_ = r.updateStatus(cfg, op.ConfigCondition{
```

Comment:
Wondering if failure in stalling add-on should mark pipeline `Config` in error state?

Reply:
yes, it should. These addons are mandatory and isn't managed by a different CR; so we have to indicate that failure to apply these addons as a failure.